### PR TITLE
Use the alias name in QuerySearchApplicationAction

### PR DESCRIPTION
**Pull request by jimczi**
The QuerySearchApplicationAction must use the alias name of the search application to build the underlying search request. This is required since using the indices name directly require the user to have the read privilege on both the alias and the underlying indices.

### CODE REVIEW

File: x-pack/plugin/ent-search/qa/rest/roles.yml [hunk 1]
```diff
@@ -22,15 +22,11 @@ admin:
...
+        "test-search-index2"
     ]
-      privileges: [ "manage", "read" ]
-    - names: [
-      # indices used for indexing and searching
-      "test-search-index1",
-      "test-search-index2",
-    ]
-      privileges: [ "manage", "read", "write" ]
+      privileges: [ "manage", "write" ]
```
Comment (carlosdelest) [inline x-pack/plugin/ent-search/qa/rest/roles.yml:29]: Are `read` privileges no longer needed for the search application alias?

File: x-pack/plugin/ent-search/qa/rest/roles.yml [hunk 2]
```diff
@@ -22,15 +22,11 @@ admin:
         "test-re-creating-search-application",
         "test-search-application-to-delete",
         "test-nonexistent-search-application",
-        "test-search-application-with-invalid-template"
+        "test-search-application-with-invalid-template",
+        "test-search-index1",
```
Comment (kderusso) [inline x-pack/plugin/ent-search/qa/rest/roles.yml:26]: Nitpick: Should we group the indices and applications together (e.g. move these two indices to line 15?)

File: x-pack/plugin/ent-search/qa/rest/roles.yml (hunk 1 above)
Comment (carlosdelest) [reply x-pack/plugin/ent-search/qa/rest/roles.yml:29]: Got it, thanks!

**Review by carlosdelest — APPROVED** [review]

**Review by kderusso — APPROVED** [review]
Good catch!

File: x-pack/plugin/ent-search/qa/rest/roles.yml (hunk 2 above)
Comment (jimczi) [reply x-pack/plugin/ent-search/qa/rest/roles.yml]: sure https://github.com/elastic/elasticsearch/pull/96176/commits/b97893eed96633cbd040f8009b0d1ee2bba89fb2

**Pull request merged**